### PR TITLE
jetty-10.0.x WebSocket core cleanup

### PR DESCRIPTION
**Pull request by lachlan-roberts**
- we no longer iterate through every extension to validate the RSV bits on a frame can be used
- `Parser` now uses live configuration from the coreSession for autoFragment and maxMessage size
- validation was cleaned up a bit in WebSocketCoreSession

### CODE REVIEW

File: jetty-websocket/websocket-core/src/main/java/org/eclipse/jetty/websocket/core/internal/ExtensionStack.java [hunk 1]
```diff
@@ -56,6 +56,7 @@
     private List<Extension> extensions;
     private IncomingFrames incoming;
     private OutgoingFrames outgoing;
+    private String[] rsvClaims = new String[3];
```
[the remark this answers is not in the document]
Comment (lachlan-roberts) [reply jetty-websocket/websocket-core/src/main/java/org/eclipse/jetty/websocket/core/internal/ExtensionStack.java]: it is set on line 198

[the remark this answers is not in the document]
Comment (gregw) [reply jetty-websocket/websocket-core/src/main/java/org/eclipse/jetty/websocket/core/internal/ExtensionStack.java]: Oh I thought it was a new field!

File: jetty-websocket/websocket-core/src/main/java/org/eclipse/jetty/websocket/core/internal/ExtensionStack.java [hunk 2]
```diff
@@ -122,8 +123,6 @@ public void negotiate(DecoratedObjectFactory objectFactory, ByteBufferPool buffe
 
         this.extensions = new ArrayList<>();
 
-        String rsvClaims[] = new String[3];
-
         for (ExtensionConfig config : negotiatedConfigs)
         {
             Extension ext;
```
Comment (gregw) [inline jetty-websocket/websocket-core/src/main/java/org/eclipse/jetty/websocket/core/internal/ExtensionStack.java]: the debug logs below (a fair way) should be protected with `if (LOG.isDebugEnabled ...`

**Comment by gregw** [comment]
@lachlan-roberts  These changes look OK, but can you check the test failures before merging.

**Pull request merged**